src/index.ts

```diff
@@ -1,7 +1,107 @@
 import { buildCSSInjectionCode, debugLog, removeLinkStyleSheets, warnLog } from './utils.js';
-import { OutputAsset, OutputChunk } from 'rollup';
-import { Plugin, ResolvedConfig } from 'vite';
-import { PluginConfiguration } from './interface';
+import type { OutputAsset, OutputBundle, OutputChunk } from 'rollup';
+import type { Plugin, ResolvedConfig } from 'vite';
+import type { PluginConfiguration } from './interface';
+
+export function extractCssAndDeleteFromBundle(bundle: OutputBundle, cssName: string): string {
+    const cssAsset = bundle[cssName] as OutputAsset;
+    const cssSource = cssAsset.source;
+    delete bundle[cssName];
+
+    // We treat these as strings and coerce them implicitly to strings, explicitly handle conversion
+    return cssSource instanceof Uint8Array ? new TextDecoder().decode(cssSource) : `${cssSource}`;
+}
+
+export function concatCss(bundle: OutputBundle, cssAssets: string[]): string {
```

Comment:
We can name it `concatCssAndDeleteFromBundle` and delete inside that method instead. So `extractCssAndDeleteFromBundle` will become simple `extractCss`